main.tf

```diff
@@ -0,0 +1,489 @@
+resource "aws_cloudwatch_log_metric_filter" "unauthorized_api_calls" {
+  count = var.unauthorized_api_calls ? 1 : 0
+
+  name           = "UnauthorizedAPICalls"
+  pattern        = "{ ($.errorCode = \"*UnauthorizedOperation\") || ($.errorCode = \"AccessDenied*\") }"
+  log_group_name = var.cloudtrail_log_group_name
+
+  metric_transformation {
+    name      = "UnauthorizedAPICalls"
+    namespace = var.alarm_namespace
+    value     = "1"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "unauthorized_api_calls" {
+  count = var.unauthorized_api_calls ? 1 : 0
+
+  alarm_name                = "UnauthorizedAPICalls"
+  comparison_operator       = "GreaterThanOrEqualToThreshold"
+  evaluation_periods        = "1"
+  metric_name               = aws_cloudwatch_log_metric_filter.unauthorized_api_calls[0].id
+  namespace                 = var.alarm_namespace
+  period                    = "300"
+  statistic                 = "Sum"
+  threshold                 = "1"
+  alarm_description         = "Monitoring unauthorized API calls will help reveal application errors and may reduce time to detect malicious activity."
+  alarm_actions             = [var.alarm_sns_topic_arn]
+  treat_missing_data        = "notBreaching"
+  insufficient_data_actions = []
+
+  tags = {
```

Comment:
Do we want to tag with env? I see that inconsistently applied. You're not otherwise using that variable, though. Just food for thought.

Reply:
This module doesn't have an `environment` variable, unless I added it by reflex? I don't see it in the `variables.tf` file though. This module is applied at the account level (because it's for Cloudtrail), so tagging with an environment doesn't make much sense.